### PR TITLE
Message Table Context Providers

### PR DESCRIPTION
**Pull request by mikedklein**
## Description

Adds to the plugin system to allow for additional MessageTable contexts.

- Additional context to the MessageTable.
- Tests for the new MessageTableProviders component.

## Motivation and Context
Need the ability to provide additional context to the MessageTable

## How Has This Been Tested?
Tested locally and unit tests have been run.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Review by roberto-graylog — APPROVED** [review]
Functionality looks good.

File: graylog2-web-interface/src/views/components/messagelist/MessageTableProviders.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import * as React from 'react';
+import { ErrorBoundary } from 'react-error-boundary';
+
+import usePluginEntities from 'views/logic/usePluginEntities';
+
+type Props = {
+  children: React.ReactElement,
+};
+
+// TODO: Rename
```
Comment (zeeklop) [inline graylog2-web-interface/src/views/components/messagelist/MessageTableProviders.tsx:26]: Is this for later?

**Review by zeeklop — APPROVED** [review]
LGTM

File: graylog2-web-interface/src/views/components/messagelist/MessageTableProviders.tsx (hunk 1 above)
Comment (mikedklein) [reply graylog2-web-interface/src/views/components/messagelist/MessageTableProviders.tsx:26]: No, I should fix that! Good catch!

**Review by simonychuang — APPROVED** [review]
LGTM!

**Review by linuspahl — APPROVED** [review]
Looks good to me!

**Pull request merged**